nvflare/job_config/script_runner.py

```diff
@@ -126,12 +129,14 @@ def add_to_fed_job(self, job, ctx, **kwargs):
                 workspace_dir="{WORKSPACE}",
             )
             metric_pipe_id = job.add_component("metrics_pipe", component, ctx)
+            comp_ids["metric_pipe_id"] = metric_pipe_id
 
             component = MetricRelay(
                 pipe_id=metric_pipe_id,
                 event_type="fed.analytix_log_stats",
             )
             metric_relay_id = job.add_component("metric_relay", component, ctx)
+            comp_ids["metric_relay_id"] = metric_relay_id
 
             component = ExternalConfigurator(
```

Comment:
maybe we add this component to comp_ids as well?

Reply:
Added. Also added additional object validation in ccwf_job.